### DISCUSSION TITLE
Show a paragraph in one line, without wrapping the text

### DISCUSSION THREAD

**Discussion by PerchunPak** [vercel/next.js#45593] [Help] [answered]
Hello!

I have a string with information, that I want to show in one line
![image](https://user-images.githubusercontent.com/68118654/216845265-f41fb95d-a61a-48cc-997b-00867c0b36dc.png)
```jsx
        <motion.h1
          className="bg-gradient-to-br from-black to-stone-500 bg-clip-text text-center font-display text-4xl font-bold tracking-[-0.02em] text-transparent drop-shadow-sm md:text-7xl md:leading-[5rem]"
          variants={FADE_DOWN_ANIMATION_VARIANTS}
        >
          <Balancer>MyMyMy Websitee</Balancer>
        </motion.h1>
        <motion.p
          className="mt-6 text-center text-gray-500 md:text-xl"
          variants={FADE_DOWN_ANIMATION_VARIANTS}
        >
          <Balancer>
            Hello there! I{"'"}m some Junior Backend/Software developer from somewhere.<br />
            Below you can find some of my main projects. They all are available on my GitHub!
          </Balancer>
        </motion.p>
```
(I have inserted some random text, to demonstrate the issue. These are changed strings from precedent.dev)

I want to show all that text in one line (like, not all in one, title should be on its own string, while this description with two strings on its own two, not 6 lines in summary as it's now).

I already tried to remove `<Balancer>` but this will save string on one line, only before last word.
![image](https://user-images.githubusercontent.com/68118654/216845505-e94852ad-5d1c-4ee9-b824-35277bda806c.png)
However, the title works as I expected.

`width: max-content` gives what I expect, except for one thing. This is not on the center now.
![image](https://user-images.githubusercontent.com/68118654/216845841-7b2e2c33-8248-4678-b408-e81753c97903.png)
(If add `text-align: center`, it shows the same result)

**Comment by icyJoseph** [comment, marked as answer]
This has to do with pure CSS. I'd like to see a repository or deployed version to play a bit with it.

Normally CSS allows text overflow, to prevent data-loss. Text overflow usually happens when, there's a word so long, that we can't just fit it in the width of the container. Otherwise, if there's no more room, the text just break at whitespaces into newlines. You can somewhat control this, with cSS.

Anyway, you've got to consider that if the viewport is too narrow, long words might cause text overflow the parent, and long sentences will wrap to fit the parent, so you gotta just accept that sometimes it won't be a exactly one line, without data-loss. 

Even if you define breakpoints and reduce text sizes, etc, users might just increasing their zoom and break the text into multiple lines. 

What I can tell you is that you probably want `width: min-content`, and most likely need to be more precise on how you center the title, this is also why I would need to see some code, the Balancer is pretty much unknown for me.

Anyhow, please do share a link to take a look.

**Reply by PerchunPak** [reply]
Thank you for the answer. I really appreciate it, but I almost fixed it in the morning with `max-width: max-content`, however it's still wrapping the last word on some pages. And as expected, on mobiles it doesn't work correctly now, I'm planning to replace `max-width: max-content` with balancer, if the page is rendered at mobile. But I don't really know how to stop wrapping at all.

If you're still interested to play, here is a patch file for https://github.com/steven-tey/precedent/commit/d82b8746c97e5754f7ba1e40b3a9f56624a505ee:

```patch
Subject: [PATCH] https://github.com/vercel/next.js/discussions/45593
---
Index: pages/index.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pages/index.tsx b/pages/index.tsx
--- a/pages/index.tsx	(revision d82b8746c97e5754f7ba1e40b3a9f56624a505ee)
+++ b/pages/index.tsx	(date 1675702145020)
@@ -42,15 +42,15 @@
           className="bg-gradient-to-br from-black to-stone-500 bg-clip-text text-center font-display text-4xl font-bold tracking-[-0.02em] text-transparent drop-shadow-sm md:text-7xl md:leading-[5rem]"
           variants={FADE_DOWN_ANIMATION_VARIANTS}
         >
-          <Balancer>Building blocks for your Next project</Balancer>
+          <Balancer>MyMyMy Websitee</Balancer>
         </motion.h1>
         <motion.p
           className="mt-6 text-center text-gray-500 md:text-xl"
           variants={FADE_DOWN_ANIMATION_VARIANTS}
         >
           <Balancer>
-            An opinionated collection of components, hooks, and utilities for
-            your Next.js project.
+            Hello there! I{"'"}m some Junior Backend/Software developer from somewhere.<br />
+            Below you can find some of my main projects. They all are available on my GitHub!
           </Balancer>
         </motion.p>
         <motion.div
```

**Reply by icyJoseph** [reply]
I mean you can try to play with:

- https://developer.mozilla.org/en-US/docs/Web/CSS/white-space
- https://developer.mozilla.org/en-US/docs/Web/CSS/word-break

And see if it lands you where you want it.
